package.json

```diff
@@ -84,6 +84,7 @@
     "watchpack": "1.6.0",
     "ws": "7.2.1",
     "yargs": "15.1.0",
+    "yargs-parser": "16.1.0",
```

Comment:
Use `yargs.Parser` to import the parser used by `yargs`, instead of adding an extra dependency.

https://github.com/yargs/yargs/blob/7293ad50d20ea0fb7dd1ac9b925e90e1bd95dea8/yargs.js#L1242-L1243

Reply:
That was my preference too, but unfortunately it doesn't play well with the webpack bundling.
When importing 'yargs' we do import yargs' `index.js` and [it doesn't export the yargs object exported by yargs.js](https://github.com/yargs/yargs/blob/7293ad50d20ea0fb7dd1ac9b925e90e1bd95dea8/index.js#L9).

I also tried to import it as "yargs/yargs" but webpack didn't liked it (it may be a recent validation added by webpack to avoid issues like [mozilla/sign-addon/pull/383](https://github.com/mozilla/sign-addon/pull/383), but I haven't double-checked if that it is actually the case).

Given that we are using the yargs-parser just to detect the command without reimplementing the parsing that yargs does internally from scratch, but we do not use it to parse any actual option, I think we may import `yargs-parser` directly (at least if I don't manage to find another way in the meantime).

Reply:
Adding `import foo from 'yargs/yargs';` results in the following error:

```
WARNING in ./node_modules/yargs/yargs.js 378:33-40
Critical dependency: require function is used in a way in which dependencies cannot be statically extracted
 @ ./src/program.js
 @ ./src/main.js
```

The error implies that webpack is attempting to bundle everything, which we obviously don't want.

To fix that, add `yargs/yargs` to `webpack.config.js`, e.g. like this:

```diff
diff --git a/webpack.config.js b/webpack.config.js
index eeb549b..dc39dba 100644
--- a/webpack.config.js
+++ b/webpack.config.js
@@ -13,6 +13,8 @@ Object.keys(require('./package.json').dependencies)
     nodeModules[mod] = 'commonjs ' + mod;
   });
 
+nodeModules['yargs/yargs'] = 'commonjs yargs/yargs';
+
 var rules = [
   {
     exclude: /(node_modules|bower_components)/,
```
With the above change, I can import and use `yargs/yargs` as expected.

Reply:
\o/ looks great to me! Thanks for looking into that part in the meantime!!!

I did applied the change needed to import the yargs-parser from yargs/yargs (and I did also run the functional tests with TEST_LEGACY_BUNDLING=1 (as we do on travis) to double-check that we are also not triggering bugs like mozilla/sign-addon/pull/383 by configuring webpack properly as in the above snippet, and the bundle is generated correctly as we thought).